docs/appendices/release-notes/5.4.4.rst

```diff
@@ -107,3 +107,8 @@ Fixes
 - Fixed an issue that caused privileges checks to be bypassed when using scalar
   sub-selects in various clauses of a query: ``SELECT``, ``WHERE``, ``HAVING``,
   etc.
+
+ - Fixed an issue that caused ``any`` operators such as ``= any``,
+   ``not like any`` or range operators like ``< any`` to throw
+   ``NullPointerExceptions`` or return invalid results when the arguments
+   contained ``nulls``.
```

Comment:
I feel like phrasing is a it confusing. After seeing `arguments` my first understanding was that array itself shouldn't be NULL but after checking examples I realized that it's a bit tricker and even single NULL inside array elements leads to NPE. 

Maybe smth like 
"...when the argument array contained at least one ``NULL``.?

Reply:
I've tested a bit myself and I don't see an issue unless there is a subquery involved:
```
cr> create table tt(a int);
CREATE OK, 1 row affected (0.328 sec)
cr> insert into tt(a) values(1), (null), (2);
INSERT OK, 3 rows affected (0.009 sec)
cr> select * from tt where a = any([1, null, 4]);
+---+
| a |
+---+
| 1 |
+---+
SELECT 1 row in set (0.004 sec)
```
```
cr> create table t(a int[]);
CREATE OK, 1 row affected (0.319 sec)
cr> insert into t(a) values([1,2,3]), ([1, null]);
INSERT OK, 2 rows affected (0.051 sec)
cr> select * from t where 1 = any(a);
+-----------+
| a         |
+-----------+
| [1, 2, 3] |
| [1, null] |
+-----------+
```
or do I miss something?

Reply:
Thank you both.

> Maybe smth like
"...when the argument array contained at least one NULL.?

updated it

> or do I miss something?

Also updated the changes entry from `= any` to `!= any`. They are handled separately as `AnyNeqOperator` and `AnyEqOperator` and `= any` did not have a problem.

Reply:
I don't see an issue with <, <=, >=, >:
```
SELECT 2 rows in set (0.002 sec)
cr> select * from t where 1 <= any(a);
+-----------+
| a         |
+-----------+
| [1, 2, 3] |
| [1, null] |
+-----------+
SELECT 2 rows in set (0.003 sec)
cr> select * from t where 1 > any(a);
+---+
| a |
+---+
+---+
SELECT 0 rows in set (0.002 sec)
cr> select * from t where 1 >= any(a);
+-----------+
| a         |
+-----------+
| [1, 2, 3] |
| [1, null] |
+-----------+
SELECT 2 rows in set (0.002 sec)
```

Reply:
Oh sorry, did not make it clear that only `AnyOperator.refMatchesAnyArrayLiteral` are fixed such that left sides must be columns.

Reply:
thx!